clang/lib/Sema/SemaStmt.cpp

```diff
@@ -290,9 +297,12 @@ void Sema::DiagnoseUnusedExprResult(const Stmt *S, unsigned DiagID) {
     if (E->getType()->isVoidType())
       return;
 
-    if (DiagnoseNoDiscard(*this, cast_or_null<WarnUnusedResultAttr>(
-                                     CE->getUnusedResultAttr(Context)),
-                          Loc, R1, R2, /*isCtor=*/false))
+    const NamedDecl *OffendingDecl;
+    const Attr *A;
+    std::tie(OffendingDecl, A) = CE->getUnusedResultAttr(Context);
```

Comment:
```suggestion
    auto &[OffendingDecl, A] = CE->getUnusedResultAttr(Context);
```

Then you can do away with the declarations.

Reply:
Oh I don't see structured binding used in this file, so I just copied the existing usage of `tie`. Will be fixed in the next push.